lib/datastore/request.js

```diff
@@ -617,11 +616,15 @@ DatastoreRequest.prototype.runQuery = function(query, callback) {
       return;
     }
 
-    var entities = entity.formatArray(resp.batch.entity_result);
+    var entities = [];
     var nextQuery = null;
 
-    if (resp.batch.end_cursor && entities.length > 0) {
-      var endCursor = resp.batch.end_cursor.toBase64();
+    if (resp.batch.entityResults) {
+      entities = entity.formatArray(resp.batch.entityResults);
+    }
+
+    if (resp.batch.endCursor && entities.length > 0) {
+      var endCursor = resp.batch.endCursor.toBase64();
 
       nextQuery = extend(true, new Query(), query);
       nextQuery.start(endCursor).offset(0);
```

Comment:
Yes, but consider this case:

    Query:
     SELECT * FROM FOO OFFSET 100
    101 things exist:
    1 -- 50
    51 - I need a break! - batch.end_cursor = pointer to this result.
    52 - 100
    101 - match!

At point 51, we should set `batch.end_cursor = pointer` *and* set `offset = 100 - skipped_results`.